src/post/post.service.ts

```diff
@@ -176,6 +183,39 @@ export class PostService {
         return this.postRepository.isRepostedByAccount(postId, accountId);
     }
 
+    private async getMentionsFromContent(content: string): Promise<Mention[]> {
+        const ctx = this.fedifyContextFactory.getFedifyContext();
+        const mentions = ContentPreparer.parseMentions(content);
+        const processedMentions: Mention[] = [];
+
+        for (const mention of mentions) {
+            let account: Account | null = null;
+            const lookupResult = await lookupActorProfile(ctx, mention);
+            if (!lookupResult.apId) {
+                continue;
+            }
+            const accountResult = await this.accountService.ensureByApId(
+                lookupResult.apId,
+            );
+            if (isError(accountResult)) {
+                continue;
+            }
+            account = getValue(accountResult);
+
+            const profileUrl = lookupResult.profileUrl
+                ? lookupResult.profileUrl
+                : lookupResult.apId;
+
+            processedMentions.push({
+                name: mention,
+                href: profileUrl,
```

Comment:
Can we use `account.url` here? This should be the profile url, and falls back to the ap id if not present

Reply:
Oh - in that case, do we even need the new `lookupActorProfile` helper? https://github.com/TryGhost/ActivityPub/blob/804e17d18653802e5b8d599cd18a0c37aecb1635/src/lookup-helpers.ts#L110C1-L179C2

Reply:
Yeah I think maybe we don't! Having said that, currently our `getApIdByHandle` lookup returns a `string` instead of a `URL` - so maybe we keep the `lookupActorProfile` and use it to return `Promise<Result<URL, LookupError>>` - and deprecate the old one? Not a big deal either way but does improve the codebase

Reply:
Thanks!! I had this same qs about the new `lookupActorProfile` helper.  
I'll create a new method that would return return `Promise<Result<URL, LookupError>>` and use it to return only `apId`.

Reply:
Did the changes in commit https://github.com/TryGhost/ActivityPub/pull/666/commits/4adb4d63b88e42c797ef80a882cf288e5f96fa1c